File: src/ClientApp/src/app/domain/entities/interfaces/IGenericRepository.ts
export interface IGenericRepository<TEntity> {
  getByIdAsync(id: number): Promise<TEntity | null>;
  getAllAsync(): Promise<TEntity[]>;
  addAsync(entity: TEntity): Promise<TEntity>;
  update(entity: TEntity): TEntity;
  delete(entity: TEntity): void;
}

File: src/Domain/Interface/IGenericRepository.cs
namespace Domain.Interface;

public interface IGenericRepository<TEntity> where TEntity : class
{
    Task<TEntity?> GetByIdAsync(int id);
    Task<IReadOnlyList<TEntity>> GetAllAsync();
    Task<TEntity> AddAsync(TEntity entity);
    TEntity Update(TEntity entity);
    void Delete(TEntity entity);
}